### PR TITLE
Add tests to make sure virtual slot is set correctly for parent types of types that have modreq members or implementations

### PR DESCRIPTION
**Pull request by OmarTawfik**
Adding a couple more tests on this corner scenarios. Basically to guard against regressions in the future where the parent method might regress to be virtual.

@jaredpar can you please confirm this meets the scenario you mentioned?
@dotnet/roslyn-compiler for review

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs [hunk 1]
```diff
@@ -4405,6 +4405,85 @@ public class Test
...
+                var parentMethod = module.ContainingAssembly.GetTypeByMetadataName("Parent").GetMethod("M");
+                Assert.False(parentMethod.IsMetadataVirtual());
+                Assert.Empty(parentMethod.Parameters.Single().RefCustomModifiers);
+
+                // Nothing on Child
+                var childMethod = module.ContainingAssembly.GetTypeByMetadataName("Child").GetMethod("M");
+                Assert.False(childMethod.IsMetadataVirtual());
+                Assert.Empty(childMethod.Parameters.Single().RefCustomModifiers);
+
+                // Modreq on Interafce
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs]: Typo, `Interface`, here and below.

File: src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs [hunk 2]
```diff
@@ -4405,6 +4405,85 @@ public class Test
...
+    public void M(in int x) { }
+}
+interface IM
+{
+    void M(in int x);
+}
+class Child: Parent, IM
+{
+    public void M(in int x) { }
+}";
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs]: It looks `Child.M` should generate a warning that `new` is required. Consider calling `VerifyDiagnostics` so it's clear that we generate the warning in this case.

File: src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs [hunk 3]
```diff
@@ -4405,6 +4405,85 @@ public class Test
...
+            {
+                // Nothing on Parent
+                var parentMethod = module.ContainingAssembly.GetTypeByMetadataName("Parent").GetMethod("M");
+                Assert.False(parentMethod.IsMetadataVirtual());
+                Assert.Empty(parentMethod.Parameters.Single().RefCustomModifiers);
+
+                // No method on Child
+                Assert.DoesNotContain("M", module.ContainingAssembly.GetTypeByMetadataName("Child").MemberNames);
+
+                // Modreq on Interafce
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs]: Nit: typo in Interface

**Comment by OmarTawfik** [comment]
@cston comments addressed.

**Pull request merged**